content/getting-started/linux.md

```diff
@@ -0,0 +1,113 @@
+
+You can instead install the full source code to the TinyGo compiler itself, generally for people who wish to contribute to the project or want to build the compiler from sources.
+
+The third option is to use the Docker image. This has the benefit of making no changes to your system, however the image has a large download and installation size.
+
+## Quick Install
+
+If you are using Ubuntu or another Debian based Linux, download the DEB file from Github and install it using the following commands:
+
+```shell
+wget https://github.com/tinygo-org/tinygo/releases/download/v0.5.0/tinygo0.5.0.linux-amd64.deb
+sudo dpkg -i tinygo0.5.0.linux-amd64.deb
+```
+
+You will need to ensure that the path to the `tinygo` executable file is in your `PATH` variable.
+
+```shell
+export PATH=$PATH:/usr/local/tinygo/bin
+```
+
+You can test that the installation is working properly by running this code which should display the version number:
+
+```shell
+$ tinygo version
+tinygo version 0.5.0 linux/amd64
+```
+
+If you are only interested in compiling TinyGo code for WebAssembly then you are done with the installation.
+
+Otherwise, please continue with the installation of the additional requirements for your desired microcontroller.
+
+### Additional Requirements for Microcontrollers
+
+There are some additional requirements to compile TinyGo programs that can run on microcontrollers.
+
+#### ARM Cortex-M
+
+To compile TinyGo programs for ARM based processors you must install CLang 8 (`clang-8`) for building assembly files and the compiler runtime library [https://compiler-rt.llvm.org/](https://compiler-rt.llvm.org/).
+
+Some boards also require a special flashing tool for that particular chip, like `openocd` or `nrfjprog`. See the documentation page for your board as listed [here](../../microcontrollers/) to see which flashing tool is required for your target board.
+
+#### AVR (Arduino)
+
+To compile and flash TinyGo programs for AVR based processors you must install some extra tools:
+
+```shell
+sudo apt-get install avr-gcc
+sudo apt-get install avr-libc
+sudo apt-get install avrdude
+```
+
+This should allow you to compile and flash TinyGo programs on an Arduino or other supported AVR-based board.
+
+## Source Install
+
+You can install TinyGo from source, which includes installing the LLVM compiler toolchain.
+
+First, obtain the TinyGo source code:
+
+```shell
+go get -d github.com/tinygo-org/tinygo
+cd $GOPATH/src/github.com/tinygo-org/tinygo
+```
+
+Once you have the code, you can install the various prerequisites using [`dep`](https://golang.github.io/dep/):
+
+```shell
+dep ensure
+```
+
+Now you can run the following make tasks to download and build the LLVM toolchain. Please note that this is likely to take at least 1 hour even on a fast machine.
+
+```shell
+make llvm-source
+make llvm-build
```

Comment:
We should probably also list build dependencies. This includes at least a C compiler (gcc, clang, generally clang is faster but needs to be set manually using `export CC=clang; export CXX=clang++`), ninja (on Debian packaged as `ninja-build`) and cmake (often just packaged as `cmake`). There are some other required things like Python and `libtinfo-dev` but most people have those already.

Reply:
Done.